main.go

```diff
@@ -26,7 +69,45 @@ func main() {
 	})
 
 	for line := range t.Lines {
-		fmt.Println(line.Text)
+		parsed := parse(line.Text)
+		if parsed != "" {
```

Comment:
You could collapse this and the above line if you felt like it:

```go
if parsed := parse(line.Text); parsed != "" {
  fmt.Println(parsed)
}
```

But there's no material benefit other than one fewer line in this case. In other cases it's useful because the variable you declare on the `if` line only exists inside of the scope of the if statement.

For example, you see this quite a lot with functions that only return an error:

```go
if err := foo(); err != nil {
  return err
}

if err := bar(); err != nil {
  return err
}
```

Note the second `:=`. That's not an error because of the way `err` gets scoped in to the if statements. If we weren't declaring `err` on the same line as the `if` the second one would be a compile-time error because we're redeclaring `err`.